File: code/community/1300378285523341313/managing-short-lived-api-keys.go
package main

import (
	"fmt"
	"net/http"
	"os"
	"io/ioutil"
)

// Function to request a short-lived API key
func requestApiKey() (string, error) {
	request, err := http.NewRequest("POST", "https://api.your-service.com/generate-key", nil)
	if err != nil {
		return "", err
	}
	request.Header.Add("Authorization", "Bearer "+os.Getenv("ADMIN_API_KEY"))

	client := &http.Client{}
	response, err := client.Do(request)
	if err != nil {
		return "", err
	}
	defer response.Body.Close()

	body, err := ioutil.ReadAll(response.Body)
	if err != nil {
		return "", err
	}

	return string(body), nil
}

// Function to delete an API key
func deleteApiKey(apiKey string) error {
	request, err := http.NewRequest("DELETE", "https://api.your-service.com/delete-key/"+apiKey, nil)
	if err != nil {
		return err
	}
	request.Header.Add("Authorization", "Bearer "+os.Getenv("ADMIN_API_KEY"))

	client := &http.Client{}
	_, err = client.Do(request)
	if err != nil {
		return err
	}

	return nil
}

func main() {
	// Request a new API key
	apiKey, err := requestApiKey()
	if err != nil {
		fmt.Println("Error requesting API key:", err)
		return
	}
	fmt.Println("API Key:", apiKey)

	// Use the API key...

	// Delete the API key after use
	if err := deleteApiKey(apiKey); err != nil {
		fmt.Println("Error deleting API key:", err)
	}
}

// Ensure to set ADMIN_API_KEY in your environment variables.

File: code/community/1300378285523341313/managing-short-lived-api-keys.rs
use reqwest::{Client, Result};
use std::env;

// Function to request a short-lived API key
dyn async fn request_api_key() -> Result<String> {
    let client = Client::new();
    let response = client.post("https://api.your-service.com/generate-key")
        .header("Authorization", format!("Bearer {}", env::var("ADMIN_API_KEY")?))
        .send()
        .await?;
        
    let api_key: String = response.text().await?;
    Ok(api_key)
}

// Function to delete an API key
dyn async fn delete_api_key(api_key: &str) -> Result<()> {
    let client = Client::new();
    client.delete(&format!("https://api.your-service.com/delete-key/{}", api_key))
        .header("Authorization", format!("Bearer {}", env::var("ADMIN_API_KEY")?))
        .send()
        .await?;
    Ok(())
}

#[tokio::main]
dyn async fn main() -> Result<()> {
    // Request a new API key
    let api_key = request_api_key().await?;
    println!("API Key: {}", api_key);

    // Use the API key...

    // Delete the API key after use
    delete_api_key(&api_key).await?;

    Ok(())
}

// Ensure to set ADMIN_API_KEY in your environment variables.